### PR TITLE
Fix EIP20Gateway naming & Use Mosaic for Contracts

### LINKED ISSUE #37: Rename Gateway -> EIP20Gateway
To keep the naming in line with the actual contract names all references of `Gateway` should be renamed to `EIP20Gateway`, and the same for `CoGateway` -> `EIP20CoGateway`.

### PR DESCRIPTION
**Pull request by hobofan**
- Make sure EIP20Gateway is used instead of Gateway in docs
- Removed SimpleToken mentions
- Adjust Contracts class to use Mosaic abstraction

FIXES #37

### CODE REVIEW

File: src/Staker/Staker.js [hunk 1]
```diff
@@ -30,8 +30,7 @@ class Staker {
   /**
    * Constructor for staker.
    *
-   * @param {Object} originWeb3 Origin chain web3 object.
-   * @param {string} Gateway contract address.
+   * @param {Object} mosaic Mosaic object.
```
Comment (schemar) [inline src/Staker/Staker.js]:
```suggestion
   * @param {Mosaic} mosaic Mosaic object.
```

**Review by schemar — APPROVED** [review]
LGTM ⛵️ 

I am not sure we still call it value token 🤔

**Review by deepesh-kn — APPROVED** [review]
LGTM :shipit:

**Comment by hobofan** [comment]
> I am not sure we still call it value token 🤔

That will be addressed in #62

**Pull request merged**